tools/plugins/com.liferay.ide.server.ui/src/com/liferay/ide/server/ui/action/RedeployAction.java

```diff
@@ -66,14 +67,21 @@ public void run( IAction action )
 
                 if( liferayServerBehavior != null )
                 {
-                    try
+                    UIUtil.async( new Runnable()
```

Comment:
but, the publish task will update the module state in server view, this don't need to run in UI thread or eclipse can handle this in background?